controllers/component_image_controller.go

```diff
@@ -0,0 +1,248 @@
+func (r *NewComponentImageReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("PipelineRun", req.NamespacedName)
+
+	log.Info("PipelineRun succeeded, checking for a new image...")
+
+	// Fetch the PipelineRun
+	var pipelineRun tektonapi.PipelineRun
+	err := r.Client.Get(ctx, req.NamespacedName, &pipelineRun)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// PipelineRun has been deleted, nothing to do.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Get Component the PipelineRun is associated to.
+	componentName := getOwnerNameOfKind(&pipelineRun, "Component")
+	if componentName == "" {
+		// Component owner is not set (usually for webhook builds), check the annotation
+		if pipelineRun.ObjectMeta.Annotations == nil || len(pipelineRun.ObjectMeta.Annotations[ComponentAnnotationName]) == 0 {
+			log.Info(fmt.Sprintf("PipelineRun '%v' has no '%s' annotation", req.NamespacedName, ComponentAnnotationName))
+			// Failed to detect Component owner, stop.
+			return ctrl.Result{}, nil
+		}
+		componentName = pipelineRun.ObjectMeta.Annotations[ComponentAnnotationName]
+	}
+
+	var component appstudiov1alpha1.Component
+	componentKey := types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      componentName,
+	}
+	err = r.Client.Get(ctx, componentKey, &component)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Component has been deleted, nothing to do.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Error(err, fmt.Sprintf("Failed to read component %v", componentKey))
+		return ctrl.Result{}, err
+	}
+
+	// Obtain newly built image
+
+	// Search for the build TaskRun
+	taskRunBelongsToPipelineRunRequirement, err := labels.NewRequirement(PipelineRunLabelName, selection.Equals, []string{pipelineRun.Name})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	taskRunPipelineTaskNameRequirement, err := labels.NewRequirement(PipelineTaskLabelName, selection.Equals, []string{BuildImageTaskName})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	componentBuildTaskRunSelector := labels.NewSelector().Add(*taskRunBelongsToPipelineRunRequirement).Add(*taskRunPipelineTaskNameRequirement)
+	var tasks tektonapi.TaskRunList
+	if err := r.Client.List(ctx, &tasks, &client.ListOptions{LabelSelector: componentBuildTaskRunSelector}); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var taskrun *tektonapi.TaskRun
+	switch len(tasks.Items) {
+	case 0:
+		// Build task not found, it was skipped
+		log.Info(fmt.Sprintf("No new image built in '%v' PipelineRun for '%v' Comoponent", req.NamespacedName, componentKey))
+		return ctrl.Result{}, nil
+	case 1:
+		taskrun = &tasks.Items[0]
+	default:
+		// Should not happen
+		log.Info(fmt.Sprintf("Found %d build tasks for %v PipelineRun", len(tasks.Items), pipelineRun))
```

Comment:
feels like we should minimally generate an k8s event for this

*maybe* a k8s metric / alert longer term

Reply:
Thanks, implemented